pkg/reconciler/testing/trigger.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"context"
+	"time"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TriggerOption enables further configuration of a Trigger.
+type TriggerOption func(*v1alpha1.Trigger)
+
+// NewTrigger creates a Trigger with TriggerOptions
+func NewTrigger(name, namespace string, so ...TriggerOption) *v1alpha1.Trigger {
+	s := &v1alpha1.Trigger{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	for _, opt := range so {
+		opt(s)
+	}
+	s.SetDefaults(context.Background())
+	return s
+}
+
+func WithSubscriberURI(uri string) TriggerOption {
```

Comment:
To avoid name conflict I have been naming these *With<Resource>TheOptionBlah*

So this should be WithTriggerSubscriberURI to avoid channel conflict.